sdk/include/opentelemetry/sdk/common/global_error_handler.h

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <atomic>
+#include <iostream>
+#include <mutex>
+
+#include "opentelemetry/common/spin_lock_mutex.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+#ifndef IGNORE_OTEL_ERROR_LOGGER
+#  define OTEL_ERROR(message) \
+    opentelemetry::sdk::common::Error::GlobalErrorHandler::GetErrorHandler()->Handle(message);
+#else
+#  define OTEL_ERROR(message)
+#endif
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+namespace Error
+{
+
+class ErrorHandler
+{
+public:
+  virtual void Handle(nostd::string_view error) noexcept = 0;
+};
+
+class DefaultErrorHandler : public ErrorHandler
+{
+public:
+  void Handle(nostd::string_view error) noexcept override
```

Comment:
I don't think this would be sufficient to cover the error cases. This would be of extremely limited value, given that you are manually instrumenting every spot with a logging macro - it could have been something better than a pointer to string.

Reply:
Can you please elaborate on the error cases it won't cover. Thanks :)

Reply:
Example: If you are posting HTTP request and collector responds with `status=400`, you need to pass the post body to error callback, so you can dump it in there and understand what's wrong with it. There could be other examples. You can find more practical examples where you need more than just a string - [here](https://github.com/microsoft/cpp_client_telemetry/search?q=DispatchEvent).